### ISSUE TITLE
Getting a 500 from service worker

### ISSUE THREAD

**Issue by voyou-sbeaulieu** [xwp/pwa-wp#184]
We just install PWA on our bedrock structure and so far everything is working well except for the service worker / offline mode.

When the page is loaded there's no error but if, from the inspect, go to the application tabs we can see the worker is running (green light). But it's not working ( no caching from application cache also ). If we hit the update link, we get this error:

A bad HTTP response code (500) was received when fetching the script.

Is there something we didn't do or forget about?
Does caching/offline work directly after activation or do we need a 3rd party etc?

Thank for your time!

**Comment by westonruter** [comment]
@voyou-sbeaulieu Is there an error in the PHP error log? And you get a 500 error when going to `/?wp_service_worker=1` in your browser? To make sure that the service worker itself isn't causing  the  error, try bypassing for the network in Chrome DevTools:

![image](https://user-images.githubusercontent.com/134745/60540860-136e5900-9cc5-11e9-96c2-94c14478eb08.png)

**Comment by voyou-sbeaulieu** [comment]
No error log, here's the URL from the test site: https://vilaincabot.voyou-web.com/

But I do have more errors from the console right now:

```logger.mjs:44 workbox Precaching 4 files. 2 files are already cached.
logger.mjs:44 workbox Navigation preload is enabled.
logger.mjs:44 workbox Router is responding to: /
logger.mjs:44 workbox Router is responding to: /
The service worker navigation preload request failed with network error: net::ERR_INTERNET_DISCONNECTED.
logger.mjs:44 workbox Router is responding to: /
The service worker navigation preload request failed with network error: net::ERR_INTERNET_DISCONNECTED.
logger.mjs:44 workbox Router is responding to: /
The service worker navigation preload request failed with network error: net::ERR_INTERNET_DISCONNECTED.
logger.mjs:44 workbox Router is responding to: /
jquery-migrate.js?ver=1.4.1:23 JQMIGRATE: Migrate is installed with logging active, version 1.4.1
logger.mjs:44 workbox Precaching is responding to: /app/uploads/2019/07/cropped-screenshot-192x192.png
A bad HTTP response code (500) was received when fetching the script.
```
Same error with "bypass for network" enable. I have this error everytime a click "Update":

`A bad HTTP response code (500) was received when fetching the script.`

**Comment by voyou-sbeaulieu** [comment]
Oh and just to add confusion, I just tried backend for fun and it's working...

We are using bedrock WordPress base and sage 9 theme. Maybe there's a conflict somewhere? 
We also did some custom adjustment/plugins...

Which info can I provide to you to help tracking this?

Thank for your time!

**Comment by voyou-sbeaulieu** [comment]
Nvm.. backend still had the offline problem. Plus there's was no error before I hit the "offline" but when I went offline, I now have `An unknown error occurred when fetching the script.` Even if I uncheck "Offline".

**Comment by westonruter** [comment]
Ok, take a look at https://vilaincabot.voyou-web.com/?wp_service_worker=1

It appears there may be a problem with how you are extending the plugin. Here is the displayed error:

```
Fatal error: Uncaught Error: Class 'App\WP_Service_Worker_Caching_Routes' not found in /home/voyou/subdomains/vilaincabot.voyou-web.com/web/app/themes/wp-boilerplate/app/filters.php:160 Stack trace: #0 /home/voyou/subdomains/vilaincabot.voyou-web.com/web/wp/wp-includes/class-wp-hook.php(286): App\{closure}('NetworkOnly') #1 /home/voyou/subdomains/vilaincabot.voyou-web.com/web/wp/wp-includes/plugin.php(208): WP_Hook->apply_filters('NetworkOnly', Array) #2 /home/voyou/subdomains/vilaincabot.voyou-web.com/web/app/plugins/pwa/wp-includes/components/class-wp-service-worker-navigation-routing-component.php(361): apply_filters('wp_service_work...', 'NetworkOnly') #3 /home/voyou/subdomains/vilaincabot.voyou-web.com/web/wp/wp-includes/class-wp-hook.php(286): WP_Service_Worker_Navigation_Routing_Component->serve(Object(WP_Service_Worker_Scripts)) #4 /home/voyou/subdomains/vilaincabot.voyou-web.com/web/wp/wp-includes/class-wp-hook.php(310): WP_Hook->apply_filters(NULL, Array) #5 /home/voyou/subdomains/vilaincabot.voyou-web.com/web/w in /home/voyou/subdomains/vilaincabot.voyou-web.com/web/app/themes/wp-boilerplate/app/filters.php on line 160
```

You may be missing a global-namespace backslash.

**Comment by voyou-sbeaulieu** [comment]
Oh thx those was some test I took from a thread to try to fix the problem but I forgot it there. I didn't saw this error from the URL you give me tho.

I remove those line and there's still something not working properly. Now when I hit "update" I got this error: `The script has an unsupported MIME type ('text/html').` and still no "offline" mode. 

Now on offline mode, I got this from the console : 
```
The service worker navigation preload request failed with network error: net::ERR_INTERNET_DISCONNECTED.
6An unknown error occurred when fetching the script.
logger.mjs:44 workbox Router is responding to: /
The service worker navigation preload request failed with network error: net::ERR_INTERNET_DISCONNECTED.
The FetchEvent for "https://vilaincabot.voyou-web.com/" resulted in a network error response: the promise was rejected.
Promise.then (async)
(anonymous) @ Router.mjs:60
?wp_service_worker=1:295 Uncaught (in promise) TypeError: Cannot read property 'text' of undefined
    at ?wp_service_worker=1:295
(anonymous) @ ?wp_service_worker=1:295
The script has an unsupported MIME type ('text/html').
``` 

I double checked and there's no more 'try to fix' code left only the plug-in installed.

Thank for your time!

**Comment by voyou-sbeaulieu** [comment]
Okie I think I find out something.

We are using bedrock base as mention before. 
So your "site_url" is `https://vilaincabot.voyou-web.com/wp/` 

I think the service worker is using this URL instead of the base "/". If I go to 
`https://vilaincabot.voyou-web.com/wp/` I see the front page instead of the normal 404 page and the offline mode IS working there... showing the home page. I also see this message : 

```
Congratulations!!Service Worker Registered ServiceWorker scope:  https://vilaincabot.voyou-web.com/wp/
``` 
Not sure what should be adjust to fit with bedrock logic tho. (https://roots.io/bedrock/)

Thank for your time

**Comment by voyou-sbeaulieu** [comment]
Ok so... My bad but those errors weren't from your plug-in. Someone in my team installed another "pwa" plug-in between yesterday testing and this morning soo... Yeah forgot about those messages I guess.

It's still not working as expected but here's the current state with your plug-in alone:

When I load the front page in an Incognito this is what the console echo:
```
JQMIGRATE: Migrate is installed with logging active, version 1.4.1
www.googletagmanager.com/gtm.js?id=GTM-TEST:1 Failed to load resource: the server responded with a status of 404 ()
logger.mjs:44 workbox Precaching 6 files.
logger.mjs:44 workbox Navigation preload is enabled.
```

No error. There's no real error from the manifest to. 

Your service working seem to use the "/" as opposed to what I saw earlier 
https://screencast.com/t/9iwwu9eF

If I hit "update" I do not get the error anymore...

When I hit the "offline" mode and then "update"  again I get an "x1" red next to the worker source but no error in the console. and the page is offline chrome default page.

```
No internet
Try:

Checking the network cables, modem, and router
Reconnecting to Wi-Fi
ERR_INTERNET_DISCONNECTED
```

Once again, sorry for the earlier confusion and thank you for your time! I lock this project to myself only so more problem like this.

**Comment by westonruter** [comment]
@voyou-sbeaulieu Make sure you uncheck “Bypass for Network”:

![image](https://user-images.githubusercontent.com/134745/60603558-d78dcf00-9d6a-11e9-8721-d37ca979250d.png)

When I do that, I am able to successfully get the default offline page served by the PWA plugin:

![image](https://user-images.githubusercontent.com/134745/60603536-ce9cfd80-9d6a-11e9-84f5-ad3a8d0759a5.png)

**Comment by alexlii1971** [comment]
Hi @westonruter,

Actually, I am quite confused by these three settings: offline , update on reload , bypass for network,  any online information for reference please?

**Comment by westonruter** [comment]
See https://developers.google.com/web/fundamentals/codelabs/debugging-service-workers/#inspecting_the_service_workers

**Comment by voyou-sbeaulieu** [comment]
Thx @westonruter ! 

So the offline page IS the "out-of-the-box" from pwa plug-in?

There's no page caching from scratch if I understand correctly?

**Comment by westonruter** [comment]
Correct. But that may change, especially after https://github.com/xwp/pwa-wp/pull/178 is merged and #176 is closed. A default network-first strategy may make sense. But otherwise it currently has to be enabled with a couple filters.

**Comment by voyou-sbeaulieu** [comment]
Okie thx I figured out you have a nice page caching example on the readme everything seems to be working fine!

Thank you for your time and support!
